.upstream-tests/test/host_only/CMakeLists.txt

```diff
@@ -9,15 +18,25 @@ function(libcudacxx_add_standalone_header_test test_file_output_var header_under
 
     set(${test_file_output_var} ${test_cpp} PARENT_SCOPE)
     # ex: cuda/std/version -> cuda_std_version
-    string(REPLACE "/" "_" executable_name ${header_under_test})
-    add_executable(${executable_name} ${test_cpp})
-    target_include_directories(${executable_name} PRIVATE ${CMAKE_SOURCE_DIR}/include)
-    target_compile_options(${executable_name} PRIVATE
-        $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>:
-            -Wall -Werror>
-        $<$<CXX_COMPILER_ID:MSVC>:
-            /W4 /WX>)
-    set_target_properties(${executable_name} PROPERTIES CXX_STANDARD 11)
+
+    foreach(version IN LISTS cpp_std_versions)
+        string(REPLACE "/" "_" executable_name "${header_under_test}-${version}")
+        list(FIND cpp_${version}_exclusions ${header_under_test} header_excluded)
+        string(FIND ${CMAKE_CXX_COMPILER} "${cpp_${version}_exclusions}" compiler_excluded)
```

Comment:
I don't think this going to do what you want.

if you had:
```
set(cpp_11_exclusions "cuda/std/mdspan" "g++")
```

It wouldn't stop when your compiler if `g++` for the given reasons:

1. You are using the full path of the compiler not the compiler id which isn't stable (  consider names like `x86_64-conda_cos6-linux-gnu-c++` )
2. You are actually asking CMake to find the string `"cuda/std/mdspan;g++" in the compiler path which won't happen.

What you want to do is use the `CMAKE_CXX_COMPILER_ID` and use `list(FIND` to see if the compiler id is one of the elements in the `cpp_${version}_exclusions` list

Reply:
Oh thanks! I actually had dropped the idea since it wasn't working. I want to drop the compiler exclusions list for now as all the tests are passing without filtering anyway since I added the `CXX_STANDARD_REQUIRED OFF` property.

Reply:
I hacked together a solution using `try_compile` instead. It seems to do exactly what I'm looking for, but seems a little heavy handed.